src/components/App.js

```diff
@@ -59,67 +86,48 @@ function App() {
           />
         </Grid>
-        <Grid item xs={6} align="left">
-          <Typography variant="p1"> Answers:</Typography>
-        </Grid>
-        <Grid item>
-          <Typography variant="ch1"> Enter Choice 1:</Typography>
-        </Grid>
-        <Grid item>
-          <OutlinedInput
-            id="outlined-adornment-password"
-            type={'text'}
-            label="Choice 1"
-            value={choice1}
-            onChange={t => {
-              setChoice1(t.target.value)
-            }}
-            endAdornment={
-              <InputAdornment position="end">
-                <FormControlLabel
-                  control={
-                    <Checkbox
-                      checked={answer1}
-                      onChange={handleAnswerCorrectnessChange}
-                      name="answer1"
-                      edge="end"
-                    />
-                  }
-                />
-              </InputAdornment>
-            }
-          />
-        </Grid>
         <Grid item>
-          <Typography variant="ch2"> Enter Choice 2:</Typography>
-        </Grid>
-        <Grid item>
-          <OutlinedInput
-            id="outlined-adornment-password"
-            type={'text'}
-            label="Choice 2"
-            value={choice2}
-            onChange={t => {
-              setChoice2(t.target.value)
-            }}
-            endAdornment={
-              <InputAdornment position="end">
-                <FormControlLabel
-                  control={
-                    <Checkbox
-                      checked={answer2}
-                      onChange={handleAnswerCorrectnessChange}
-                      name="answer2"
-                      edge="end"
-                    />
-                  }
-                />
-              </InputAdornment>
-            }
-          />
+          <Typography variant="p1"> Answers:</Typography>
         </Grid>
+        {choices.map((choice, index) => (
+          <div key={index}>
+            <Grid container direction={"column"} columns={{ xs: 12, sm: 12, md: 12, lg: 12, xl: 12 }} align="left">
+              <Grid item sx={{ pt: 2 }}>
+                <Typography> Enter Choice {index + 1}:</Typography>
+              </Grid>
+              <Grid item variant="outlined" sx={{ pt: 2 }}>
+                <FormControl variant="outlined">
+                  <InputLabel >Choice {index + 1}</InputLabel>
+                  <OutlinedInput
+                    type={'text'}
+                    label={`Choice ${index + 1}`}
+                    value={choice}
+                    placeholder={`Enter Choice ${index + 1}`}
+                    onChange={(e) => handleChoiceChange(index, e)}
+                    endAdornment={
+                      <InputAdornment position="end">
+                        <FormControlLabel
+                          control={
+                            <Checkbox
+                              checked={answers[index].name}
```

Comment:
I don't know why you use name here. 
With my above solution, it would be `choices[index].isCorrect`